### ISSUE TITLE
Flocking doesn't output audio on Node.js on Windows 10

### ISSUE THREAD

**Issue by colinbdclark** [colinbdclark/Flocking#179]
No audio is output when running Flocking in Node.js on Windows 10 (and possibly other versions, I haven't tested). No errors are thrown; there's just no sound.

With some debugging, this is due to a bug in that was [recently fixed upstream in node-speaker](https://github.com/TooTallNate/node-speaker/pull/74), the library used by Flocking to output audio. This change, in turn, caused Flocking to regress on Windows, since it appears that node-speaker now does not support samples in 32-bit floating point format. This latter issue is [fixed in a pending pull request](https://github.com/TooTallNate/node-speaker/pull/79) to node-speaker that apparently hasn't been reviewed yet.

After this, we'll still need to update our low-latency hack branch of node-speaker until they provide some means of surfacing mpg123 buffer sizes to JavaScript. There's also an unfinished pull request to node-speaker for this, but it's been dormant for a while.

The bottom line: we need a more reliable audio backend in Node.js, preferably one based on portaudio.

**Comment by colinbdclark** [comment]
[Fixed upstream in my fork of node-speaker](https://github.com/colinbdclark/node-speaker/tree/mpg123-lower-latency) by merging in the [latest master](https://github.com/colinbdclark/node-speaker/commit/d5fe7b388495783969a07369c7b9744c008cb705) and the [above-mentioned pull request](https://github.com/colinbdclark/node-speaker/commit/41f80749d7fe3eea2c4c391c9330383cf537dd36).
